test/test_jit.py

```diff
@@ -1396,6 +1396,37 @@ def constant_prop():
         self.run_pass('constant_propagation', constant_prop.graph)
         self.assertExpected(canonical(constant_prop.graph))
 
+    def test_trace_detach(self):
+        def foo(x, w):
+            return torch.matmul(x, w).detach()
+
+        traced = torch.jit.trace(torch.rand(3, 4), torch.rand(4, 5))(foo)
+
+        self.assertExpected(str(traced.graph))
+        x, w = torch.rand(3, 4), torch.rand(4, 5)
+        self.assertEqual(foo(x, w), traced(x, w))
```

Comment:
Are you sure `assertEqual` checks the `requires_grad` status? I'm pretty sure it only cares about the data, so you need to do it yourself